src/data/list/basic.lean

```diff
@@ -1327,6 +1327,17 @@ lemma last_eq_nth_le : ∀ (l : list α) (h : l ≠ []),
 | (a :: b :: l) h := by { rw [last_cons, last_eq_nth_le (b :: l)],
                           refl, exact cons_ne_nil b l }
 
+lemma nth_le_length_sub_one {l : list α} (h : l.length - 1 < l.length) :
+  l.nth_le (l.length - 1) h = l.last (by { rintro rfl, exact nat.lt_irrefl 0 h }) :=
+(last_eq_nth_le l _).symm
+
+lemma nth_le_length_cons {l : list α} (a : α) (h : l ≠ []) :
```

Comment:
We already have `list.nth_le_cons_length`, which has a better name and is stated without needing `h`.

Reply:
Thanks!